db/repos/accounts.js

```diff
@@ -233,33 +233,22 @@ class AccountsRepository {
 			); // eslint-disable-line prefer-promise-reject-errors
 		}
 
-		if (conflictingFields.length === 1 && conflictingFields[0] === 'address') {
-			const sql =
-				'${insertSQL:raw} ON CONFLICT(${conflictFields:name}) DO UPDATE SET ${setsSQL:raw}';
-
-			return this.db.none(sql, {
-				insertSQL: this.pgp.helpers.insert(data, this.cs.insert),
-				conflictFields: conflictingFields,
-				setsSQL: this.pgp.helpers.sets(updateData, this.cs.update),
-			});
-		} else {
-			const conditionObject = {};
-			conflictingFields.forEach(function(field) {
-				conditionObject[field] = data[field];
-			});
+		const conditionObject = {};
+		conflictingFields.forEach(function(field) {
+			conditionObject[field] = data[field];
+		});
 
-			return this.db.tx('db:accounts:upsert', function(t) {
-				return t.accounts
-					.list(conditionObject, ['address'])
-					.then(function(result) {
-						if (result.length) {
-							return t.accounts.update(result[0].address, updateData);
-						} else {
-							return t.accounts.insert(data);
-						}
-					});
-			});
-		}
+		return this.db.tx('db:accounts:upsert', function(t) {
```

Comment:
This could probably be improved by an `UPSERT` logic, but for now at least this one looks better ;)

```js
    return this.db.tx('db:accounts:upsert', function* (t) {
        const result = yield t.accounts.list(conditionObject, ['address']);
        if (result.length) {
            yield t.accounts.update(result[0].address, updateData);
        } else {
            yield t.accounts.insert(data);
        }
    });
```

Reply:
It is better because you do not need to return things you don't care about, like the results from `insert/update` ;)

Reply:
@vitaly-t Yes initially we had that `INSERT .. ON CONFLICT UPDATE ...` SQL, but the reason I have to commit it; 

1. It requires a index column for the conflict attributes 
2. Our modules and logic layer are calling this repo method for account only with address

So to generateting above query with only 1 attribute was causing syntax error, as address is not in update column set. 

Also we have to manage two code blocks for one purpose, so after discussing with @4miners we decide to just keep one for now.